packages/backend/src/controllers/courseList/courseList.ts

```diff
@@ -0,0 +1,118 @@
+import { CourseList, ICourseListSchema } from './courseList.schema';
+import { fetchEnrolledCourses } from '../moodle/fetch';
+import { loggerFile } from '../../configuration/logger';
+import { config } from '../../configuration/environment';
+import { getBaseUrl } from '../moodle/index';
+import { ApiError } from '../error/api.class';
+import { Request, Response, NextFunction } from 'express';
+import { object, number, boolean } from '@hapi/joi';
+
+
+/**
+ * Creates a blacklist for excluding some courses from notification.
+ * @returns A list of course IDs for courses where no notification is desired.
+ */
+export async function getCourseBlacklist():Promise<number[]>{
+    let dbCourses = await CourseList.find();
+    if (!dbCourses) dbCourses = await getCourseList(); // Create database entrys if there is nothing yet
+    const result:number[] = [];
+    dbCourses.forEach(element => {
+        if (!element.isActive) result.push(element.courseId);
+    });
+    return result;
+}
+
+/**
+ * Fetches all moodle courses and compares them with the database.
+ * @returns A List of objects with course name, id and active status.
+ */
+export async function getCourseList():Promise<ICourseListSchema[]>{
+    const moodleCourses = await fetchEnrolledCourses(getBaseUrl());
+    const result:any[] = [];
+    for (const element of moodleCourses){
+        const courseName = (config.moodle.useCourseShortname ? element.shortname : element.fullname);
+        let dbCourse = await CourseList.findOne({courseId: element.id});
+        if (!dbCourse) dbCourse = await new CourseList({courseId: element.id, name: courseName, isActive: true}).save();
+        result.push({courseId: dbCourse.courseId, name: dbCourse.name, isActive: dbCourse.isActive});
+    }
+    return result;
+}
+
+/**
+ * Handles GET /api/settings/courses requests and responds
+ * with an array of course objects of type ICourseListSchema.
+ * @param req Request
+ * @param res Response
+ * @param next NextFunction
+ */
+export async function getCourseListRequest(req: Request, res: Response, next: NextFunction) {
+
+    try {
+        const courses = await getCourseList();
+        if (courses.length === 0) throw new ApiError(404, 'No courses found');
+        res.status(200).json(courses);
+
+    }
+    catch (err) {
+        loggerFile.error(err.message);
+        next(err);
+    }
+  }
+
+/**
+ * Method for setting a course active or inactive
+ * @param courseId ID of the course to be modified (number)
+ * @param active Boolean for setting the course to active or not
```

Comment:
```suggestion
 * @param courseId {number} ID of the course to be modified
 * @param active {boolean} Boolean for setting the course to active or not
```
Gut möglich, dass ich mich hierbei irre, aber ich glaube der Typ sollte so angegeben werden (das selbe gilt auch für die anderen Kommentare)